oak_attestation_verification/src/verifier.rs

```diff
@@ -372,382 +388,537 @@ fn verify_amd_sev_attestation_report(
             // Right now there are only Milan CPUs, so it is not urgent to code the
             // decision between Milan and Genoa which would appear here.
             let ask_milan = Certificate::from_pem(ASK_MILAN_CERT_PEM)
                 .map_err(|_err| anyhow::anyhow!("could not parse ASK cert"))?;
+
+            // TODO(#4747): user current date as part of VCEK verification.
             verify_cert_signature(&ask_milan, &vcek)?;
 
-            verify_amd_sev_attestation_report(&l.remote_attestation_report, &vcek, amd_sev)?
+            let report = AttestationReport::ref_from(&root_layer.remote_attestation_report)
+                .context("invalid AMD SEV-SNP attestation report")?;
+            report.validate().map_err(|msg| anyhow::anyhow!(msg))?;
+
+            // Ensure that the attestation report is signed by the VCEK public key.
+            verify_attestation_report_signature(&vcek, report)
         }
-        TeePlatform::IntelTdx => {
-            let intel_tdx = r
+        TeePlatform::IntelTdx => anyhow::bail!("not supported"),
+    }
+}
+
+/// Verifies the measurement values of the root layer containing the attestation report.
+fn verify_root_layer(
+    _now_utc_millis: i64,
+    values: &RootLayerValues,
+    _endorsements: Option<&RootLayerEndorsements>,
+    reference_values: &RootLayerReferenceValues,
+) -> anyhow::Result<()> {
+    match values.report.as_ref() {
+        Some(Report::SevSnp(report_values)) => verify_amd_sev_attestation_report(
+            report_values,
+            reference_values
+                .amd_sev
+                .as_ref()
+                .context("AMD SEV-SNP reference values not found")?,
+        ),
+        Some(Report::Tdx(report_values)) => verify_intel_tdx_attestation_report(
+            report_values,
+            reference_values
                 .intel_tdx
                 .as_ref()
-                .ok_or_else(|| anyhow::anyhow!("no Intel TDX reference values"))?;
-            verify_intel_tdx_attestation_report(&l.remote_attestation_report, intel_tdx)?
-        }
+                .context("Intel TDX reference values not found")?,
+        ),
+        None => Err(anyhow::anyhow!("no attestation report")),
     }
-
-    let e_stage0 = e
-        .stage0
-        .as_ref()
-        .ok_or_else(|| anyhow::anyhow!("no stage0 endorsement"))?;
-    let r_stage0 = r
-        .amd_sev
-        .as_ref()
-        .ok_or_else(|| anyhow::anyhow!("no amd_sev in root reference values"))?
-        .stage0
-        .as_ref()
-        .ok_or_else(|| anyhow::anyhow!("no stage0 reference value"))?;
-
-    // WIP: Parse attestation report and verify stage0 measurement.
-    verify_transparent_release_endorsement(now_utc_millis, e_stage0, r_stage0)
 }
 
-/// Verifies all ingredients of the kernel layer, which is common to both
-/// Oak Restricted Kernel and Oak Containers setups.
+/// Verifies the measurement values of the kernel layer, which is common to both the  Oak Restricted
```

Comment:
I used to stay below 80 cols per line, not sure what's our limit. Google3 has none so far (i.e. there is no style guide so far).

Reply:
The rustfmt configuration we use on the repository uses 100 cols.